src/crimson/osd/backfill_state.cc

```diff
@@ -271,10 +276,22 @@ BackfillState::Enqueuing::update_on_peers(const hobject_t& check)
       // because their peer_bi.begin > backfill_info.begin.
       if (primary_bi.begin > peering_state().get_peer_last_backfill(bt) &&
           backfill_state().progress_tracker->enqueue_push(primary_bi.begin)) {
-        backfill_listener().enqueue_push(primary_bi.begin, obj_v);
+	auto &v = backfills[primary_bi.begin].first;
+	auto &peers = backfills[primary_bi.begin].second;
+	assert(v == obj_v || v == eversion_t());
+	v = obj_v;
+	peers.push_back(bt);
       }
     }
   }
+  for (auto &backfill : backfills) {
+    auto &soid = backfill.first;
+    auto &obj_v = backfill.second.first;
+    auto &peers = backfill.second.second;
+    logger().debug("{}: pushing {},{} to {}", __func__, soid, obj_v, peers);
+    peering_state().prepare_backfill_for_missing(soid, obj_v, peers);
```

Comment:
Why to not make this a part of the `BackfillListener::enqueue_push()`?

Reply:
Yeah, will change it.